RxSwiftWorkshop/Flights/Model/Flight.swift

```diff
@@ -0,0 +1,131 @@
+        self.number = number
+        self.schedule = schedule
+        self.prefixIATA = prefixIATA
+        self.serviceType = serviceType
+        self.mainFlight = mainFlight
+        self.codeshares = codeshares
+        self.terminal = terminal
+        self.gate = gate
+        self.estimatedLandingTime = estimatedLandingTime
+        self.actualLandingTime = actualLandingTime
+        self.publicEstimatedOffBlockTime = publicEstimatedOffBlockTime
+        self.actualOffBlockTime = actualOffBlockTime
+        self.expectedTimeGateOpen = expectedTimeGateOpen
+        self.expectedTimeBoarding = expectedTimeBoarding
+        self.expectedTimeGateClosing = expectedTimeGateClosing
+        self.statuses = statuses
+        self.destinationCodes = destinationCodes
+    }
+
+    init(object: MarshaledObject) throws {
+        id = try object.value(for: "id")
+        number = try object.value(for: "flightNumber")
+        name = try object.value(for: "flightName")
+
+        let scheduleDate: String = try object.value(for: "scheduleDate")
+        let scheduleTime: String = try object.value(for: "scheduleTime")
+        if let dateInRegion = DateInRegion(string: "\(scheduleDate)T\(scheduleTime)", format: .iso8601Auto) {
+            schedule = dateInRegion.absoluteDate
+        } else {
+            schedule = Date()
+        }
+
+        prefixIATA = try object.value(for: "prefixIATA")
+        serviceType = try object.value(for: "serviceType")
+        mainFlight = try object.value(for: "mainFlight")
+        codeshares = (try? object.value(for: "codeshares.codeshares")) ?? []
+
+        terminal = try object.value(for: "terminal")
+        gate = try object.value(for: "gate")
+
+        estimatedLandingTime = try? object.value(for: "estimatedLandingTime")
+        actualLandingTime = try? object.value(for: "actualLandingTime")
+        publicEstimatedOffBlockTime = try? object.value(for: "publicEstimatedOffBlockTime")
+        actualOffBlockTime = try? object.value(for: "actualOffBlockTime")
+
+        expectedTimeGateOpen = try? object.value(for: "expectedTimeGateOpen")
+        expectedTimeBoarding = try? object.value(for: "expectedTimeBoarding")
+        expectedTimeGateClosing = try? object.value(for: "expectedTimeGateClosing")
+
+        statuses = try object.value(for: "publicFlightState.flightStates")
+        destinationCodes = try object.value(for: "route.destinations")
+        destinations = nil
+    }
+
+    func with(destinations: [Airport]) -> Flight {
+        var flight = self
+        flight.destinations = destinations
+        return flight
+    }
+}
+
+func ==(lhs: Flight, rhs: Flight) -> Bool {
+    return lhs.id == rhs.id
+        && lhs.name == rhs.name
+        && lhs.number == rhs.number
+        && lhs.schedule == rhs.schedule
+        && lhs.prefixIATA == rhs.prefixIATA
+        && lhs.serviceType == rhs.serviceType
+        && lhs.mainFlight == rhs.mainFlight
+        && lhs.statuses == rhs.statuses
+}
```

Comment:
There's a lot of properties not compared here. Maybe we could either update them or revert to the `dump` hack?